### PR TITLE
[FIX] fieldservice_stock: Move Direction to Stock Request

### PR DESCRIPTION
**Pull request by osi-scampbell**
This PR we take out the direction field because it is being added to OCA/stock-logistics-warehouse/stock_request here: 
- [x] https://github.com/OCA/stock-logistics-warehouse/pull/675
- [x] https://github.com/OCA/stock-logistics-warehouse/pull/606

### CODE REVIEW

File: fieldservice_stock/models/stock.py [hunk 1]
```diff
@@ -10,11 +10,6 @@ class StockRequest(models.Model):
...
         'fsm.order', string="FSM Order", ondelete='cascade',
         index=True, copy=False)
-    direction = fields.Selection([('outbound', 'Outbound'),
-                                  ('inbound', 'Inbound')],
-                                 string='Direction',
-                                 states={'draft': [('readonly', False)]},
-                                 readonly=True)
 
     @api.onchange('direction', 'fsm_order_id')
     def _onchange_location_id(self):
```
Comment (max3903) [inline fieldservice_stock/models/stock.py]: @osi-scampbell Should include a call to `super()`

File: fieldservice_stock/models/stock_request.py [hunk 2]
```diff
@@ -57,10 +57,17 @@ def create(self, vals):
...
+            res.update({
+                'fsm_order_id': self.fsm_order_id.id,
+                'partner_id': self.fsm_order_id.location_id.
+                shipping_address_id.id,
+            })
+        else:
+            res.update({
+                'fsm_order_id': self.fsm_order_id.id,
+                'partner_id': self.fsm_order_id.location_id.partner_id.id,
+            })
```
Comment (max3903) [inline fieldservice_stock/models/stock_request.py:77]: @osi-scampbell You can do:

```
if self.fsm_order_id:
    res.update({
        'fsm_order_id': self.fsm_order_id.id,
        'partner_id':
            self.fsm_order_id.location_id.shipping_address_id.id
            or self.fsm_order_id.location_id.partner_id.id,
    })
```

Comment (osi-scampbell) [reply fieldservice_stock/models/stock_request.py:77]: Got it, implementing now.

**Review by max3903 — APPROVED** [review]

File: fieldservice_stock/models/stock_request.py [hunk 3]
```diff
@@ -1,10 +1,105 @@
...
+
+    @api.model
+    def create(self, vals):
+        if 'fsm_order_id' in vals and vals['fsm_order_id']:
+            fsm_order = self.env['fsm.order'].browse(vals['fsm_order_id'])
+            fsm_order.request_stage = 'draft'
+
+            val_date = vals['expected_date']
+            if not isinstance(vals['expected_date'], str):
+                val_date = datetime.strftime(vals['expected_date'], '%Y-%m-%d %H:%M:%S')
```
Comment (bodedra) [inline fieldservice_stock/models/stock_request.py]:
```suggestion
                val_date = datetime.strftime(vals['expected_date'],
                                             '%Y-%m-%d %H:%M:%S')
```

File: fieldservice_stock/__manifest__.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
     'website': 'https://github.com/OCA/field-service',
     'depends': [
         'fieldservice',
+        'stock_request_direction',
```
Comment (bodedra) [inline fieldservice_stock/__manifest__.py:15]:
```suggestion
        'fieldservice_geoengine',
        'stock_request_direction',
```

**Review by bodedra — COMMENTED** [review]
It will make CI happy.

File: fieldservice_stock/__manifest__.py (hunk 4 above)
Comment (max3903) [reply fieldservice_stock/__manifest__.py:15]: It will not solve the Travis error. See https://github.com/OCA/geospatial/issues/206

**Review by bodedra — APPROVED** [review]
LGTM :+1:

**Pull request merged**